### ISSUE TITLE
Date parser quirks

### ISSUE THREAD

**Issue by mikolajpp** [urbit/arvo#780]
This seems like genuine bug:
the relative/absolute date parser in hoon.hoon does not limit fractional seconds to 8 bytes, but 
then attempting to type more results with decrement-underflow and dojo reset. 

How to reproduce:
```
dojo> ~s0..dead.beef.dead.beef
```
After typing above line try to append some more bytes. It will end up with:
```
/~rolled-latdep-ponmep-litsem/home/0/app/dojo:<[1.391 13].[1.391 50]>
decrement-underflow
[ %drum-coup-fail
  ~rolled-latdep-ponmep-litsem
  1
  p=~rolled-latdep-ponmep-litsem
  q=%dojo
]
/~rolled-latdep-ponmep-litsem/home/0/app/dojo:<[1.391 13].[1.391 50]>
decrement-underflow
[ %drum-coup-fail
  ~rolled-latdep-ponmep-litsem
  1
  p=~rolled-latdep-ponmep-litsem
  q=%dojo
]
```

**Comment by mikolajpp** [comment]
The bug seems to be here:
```
++  yule                                                ::  time atom
  |=  rip/tarp
  ^-  @d
  =+  ^=  sec  ;:  add
                 (mul d.rip day:yo)
                 (mul h.rip hor:yo)
                 (mul m.rip mit:yo)
                 s.rip
               ==
  =+  ^=  fac  =+  muc=4
               |-  ^-  @
               ?~  f.rip
                 0
               =>  .(muc (dec muc))
               (add (lsh 4 muc i.f.rip) $(f.rip t.f.rip))
  (con (lsh 6 1 sec) fac)
```
`++tarp` keeps a list of fractions, but in reality it should have at most 4 elements, 
which is not checked. For longer lists `(dec muc)` will underflow

**Comment by cgyarvin** [comment]
Is genuine bug.

**Comment by mikolajpp** [comment]
Ok, do I submit PR here or in research branch? It is about time to earn that "contributor" badge ;-)

**Comment by cgyarvin** [comment]
Master PR will get worked in fine.

**Comment by mikolajpp** [comment]
I have discovered several more of this. As it is very unlikely that someone will hit this, how about we wait 
until I port the full parser once complete, instead of doing micro fixes here and there...

**Comment by mikolajpp** [comment]
I found what I think is undocumented date syntax, did not find this anywhere.
`++date` has a boolean field which specifies which way we count from era, specified using a hyphen. 

Example:
~2018-.8.16 
will count from ERA downward, while 
~2018.8.16 will count from ERA upward

This means that actual year becomes
year = ERA - (year - 1) in the first case and
year = ERA + year 

initially i thought this has something to do with the leap year, but 
this is not so; Except that the mapping is degenerate as year
`1-.x.x` is the same as 0.x.x, I can't understand 
what is the use for this additional date syntax

**Comment by cgyarvin** [comment]
Yeah I think it’s usually called BC

**Comment by mikolajpp** [comment]
Uh, I got misled by the magic year again! This entire time I was thinking about ERA being already in a distant past, and everything relative to it forward in time, meanwhile it is still relative to 0AD

Do we have docs on this?

**Comment by cgyarvin** [comment]
Josh, Rob?

**Comment by rmariani** [comment]
I do not believe that this is documented anywhere.  I will make a note to document it.

**Comment by mikolajpp** [comment]
@rmariani 

I have already got documentation on this. Ping me on ~ponmep-litsem, will send you

**Comment by joshuareagan** [comment]
The right answer is probably an aura reference page that aspires to complete coverage of all custom literal syntax.
